### PR TITLE
Fix #34 for ballot measures

### LINKED ISSUE #34: Include Form 496 data in ballot measure candidate / opposition totals
This form lists independent expenditures in support or opposition to a ballot measure or candidate.

Currently there is not much here except for some school advocacy, but this sheet is gonna fill up soon.

- [ ] ballot measure supporting / opposing calculator
- [ ] ? candidate opposing calculator ( 🌟 NEW!)

Matching up to candidate is going to be tricky since the form doesn't include the candidate's recipient committee id :(. For any ballot measures without a Bal_Num we can use, add to the mapping in `models/oakland_referendum.rb`.

### PR DESCRIPTION
**Pull request by mikeubell**
This contains fixes for #34 and #42

### CODE REVIEW

File: calculators/referendum_supporters_calculator.rb [hunk 1]
```diff
@@ -6,11 +6,19 @@ def initialize(candidates: [], ballot_measures: [], committees: [])
...
         SUM("Amount") AS "Total_Amount"
-      FROM "efile_COAK_2016_E-Expenditure"
-      WHERE "Bal_Name" IS NOT NULL
+      FROM (
+        SELECT "Filer_ID", "Filer_NamL", "Bal_Name", "Sup_Opp_Cd", "Amount"
+        FROM "efile_COAK_2016_E-Expenditure"
+        WHERE "Bal_Name" IS NOT NULL
+        UNION
+        SELECT "Filer_ID"::varchar, "Filer_NamL", "Bal_Name", "Sup_Opp_Cd", "Amount"
+        FROM "efile_COAK_2016_496"
```
Comment (tdooner) [inline calculators/referendum_supporters_calculator.rb]: I am pretty sure that 496 is like form 497 in that you *always* have to add a `WHERE` "Form_Type" argument to only get expenditures/contributions.

I think you want: `WHERE "Form_Type" = 'F496P1'`

Comment (mikeubell) [inline calculators/referendum_supporters_calculator.rb]: The data that is there does not seem to support this. The only value in FormType is 'F496'

Comment (sfdoran) [inline calculators/referendum_supporters_calculator.rb]: Looking at the data they are all expenditures, so I can't tell if there's a different entry for contributions. I'll check the data key.

Comment (sfdoran) [inline calculators/referendum_supporters_calculator.rb]: @mikeubell According to the data key for Form_Type, "F496 = independent expenditures made" https://data.sfgov.org/City-Management-and-Ethics/Campaign-Finance-Data-Key/wygs-cc76

Comment (sfdoran) [inline calculators/referendum_supporters_calculator.rb]: Form_Type = F496P3 is a receipt.

Comment (mikeubell) [inline calculators/referendum_supporters_calculator.rb]: So it is always "F496" there is no other value.  Looks like contributions get put in efile_COAK_2016_F496P3-Contributions.

**Comment by mikeubell** [comment]
I have repushed, after reworking the query.  Thinking about what @tdooner said about UNION not working right, I agree, particularly in this case where there is no Exp_Code in the 496 data.

**Comment by mikeubell** [comment]
I have added code for expenses opposing candidates.  Currently there are none.  So the code is probably suspect.
I also realized that my duplicate check in the 496s did not need to look for matching ballot measures since we are only doing candidates here.

**Pull request merged**